### ISSUE TITLE
Warn or error if `builtin` or module is assigned too

### ISSUE THREAD

**Issue by PetrochukM** [microsoft/pyright#1155]
**Is your feature request related to a problem? Please describe.**
`Pyright` does not error if a module or `builtin` is assigned to. For example, the code below produced no errors...

```python3
import pathlib

str = "foo"
slice = tuple()
list = "?"
pathlib.Path() # Ensure `pathlib` is used before assigned to
pathlib = "boo"
```

**Describe the solution you'd like**
The type checking should be extended to handle `builtin`s and modules. For example, since `pathlib` is a module, it should produce an error, if I try to assign "boo" to it. It should also error if I assign "?" to `list`.

**Comment by erictraut** [comment]
An import statement creates a new local symbol and assigns the imported symbol to it. This isn't considered a type declaration, so it's legitimate from the perspective of the type checker to overwrite it with another value of another type, just as it is with other symbols without type declarations.

For example:
```python
foo = "hi"
foo = 3 # this is fine

bar: str = "hi"
bar = 3 # this is not because bar has a declared type of str
```

Likewise, this is legitimate:
```python
from x import y as foo
foo = 3
```

as is
```python
from x import foo
foo = 3
```

The ability to reassign imported symbols comes up quite frequently within Python code. For example:
```python
try:
   from x import max_vector_size
else:
  max_vector_size = 3
```

If pyright were to treat modules or imported symbols as having declared types, it would create many false positives in typical Python code.

Built-in symbols are implicit import statements, and the local symbols introduced into a module's namespace through these import statements are treated the same as other (explicit) imports.

**Comment by PetrochukM** [comment]
Okay, that's fair, even so, I think it should be a warning. It's not pythonic to redefine built-in functions. Do you know of a tool that could help me check for issues like this one? I'd love to have something similar to this rule in `eslint`: https://eslint.org/docs/rules/no-shadow
